lib/Persistence/AbstractPersistence.php

```diff
@@ -82,21 +82,23 @@ protected static function _exists($filename)
      */
     protected static function _initialize()
     {
-        // Create storage directory if it does not exist.
-        if (!is_dir(self::$_path)) {
-            if (!@mkdir(self::$_path, 0700)) {
-                throw new Exception('unable to create directory ' . self::$_path, 10);
+        if (property_exists($data->meta, 'webserver') && $data->meta->webserver && $this->_conf->getKey('webserver') == "Apache") {
+            // Create storage directory if it does not exist.
+            if (!is_dir(self::$_path)) {
+                if (!@mkdir(self::$_path, 0700)) {
+                    throw new Exception('unable to create directory ' . self::$_path, 10);
+                }
             }
-        }
-        $file = self::$_path . DIRECTORY_SEPARATOR . '.htaccess';
-        if (!is_file($file)) {
-            $writtenBytes = @file_put_contents(
-                $file,
-                'Require all denied' . PHP_EOL,
-                LOCK_EX
-            );
-            if ($writtenBytes === false || $writtenBytes < 19) {
-                throw new Exception('unable to write to file ' . $file, 11);
+            $file = self::$_path . DIRECTORY_SEPARATOR . '.htaccess';
+            if (!is_file($file)) {
+                $writtenBytes = @file_put_contents(
```

Comment:
It would be awesome if you could also detect the server version of Apache as Apache <2.4 needed a different format. So if you want to solve https://github.com/PrivateBin/PrivateBin/issues/194#issuecomment-289106771.

Hint: https://github.com/PrivateBin/PrivateBin/commit/ce92bfa934fd4df5d517216f31fe8c6aa97d51ff#diff-bb97f7ec0a7c6ca7a5f672f80703358cL307